src/core/Akka/Dispatch/ActorTaskScheduler.cs

```diff
@@ -158,18 +158,21 @@ public static void RunTask(Func<Task> asyncAction)
                               .Unwrap()
                               .ContinueWith(parent =>
                               {
-                                  Exception exception = GetTaskException(parent);
-
+                                  var exception = GetTaskException(parent);
                                   if (exception == null)
                                   {
                                       dispatcher.Resume(context);
-
                                       context.CheckReceiveTimeout();
                                   }
                                   else
                                   {
                                       context.Self.AsInstanceOf<IInternalActorRef>().SendSystemMessage(new ActorTaskSchedulerMessage(exception, actorScheduler.CurrentMessage));
                                   }
+                                  
+                                  // Used by TestActorRef to intercept async execution result
+                                  if(actorScheduler is IAsyncResultInterceptor interceptor)
```

Comment:
I'd prefer not to do this on the `TaskScheduler` that all actors use, but I don't really see a way around it - this is probably the best call we can make in the short run, without redesigining the mailbox system to use `Task`.

Reply:
Yes, other alternatives would require a more radical change to implement, this is the shortest solution I can find without scrambling codes all over the place